Library/Sender.ts

```diff
@@ -327,31 +347,50 @@ class Sender {
         );
     }
 
+    private _logInfo(message?: any, ...optionalParams: any[]) {
+        if (!this._isStatsbeatSender) {
+            Logging.info(Sender.TAG, message, optionalParams);
+        }
+    }
+
+    private _logWarn(message?: any, ...optionalParams: any[]) {
+        if (!this._isStatsbeatSender) {
+            Logging.warn(Sender.TAG, message, optionalParams);
+        }
+    }
+
+    private _statsbeatFailedToIngest() {
+        this._failedToIngestCounter++;
```

Comment:
should you only increment when _shutdownStatsbeat is not null? what if customer failed 3 times, then statsbeat failed, shutdown is not null, it will shutdown on 1st occurrence.